torch/csrc/distributed/autograd/context/container.h

```diff
@@ -117,7 +117,7 @@ class TORCH_API DistAutogradContainer {
     std::unordered_map<int64_t, ContextPtr> contexts;
   };
 
-  DistAutogradContainer();
+  DistAutogradContainer() = delete;
```

Comment:
Wow, this is neat. So before that it was up to linker to check that DistAutogradContainer is not constructed by default? 8)